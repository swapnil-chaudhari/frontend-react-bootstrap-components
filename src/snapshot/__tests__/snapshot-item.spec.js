import React from 'react';
import renderShallow from 'render-shallow';
import SnapshotItem from '../snapshot-item';

describe('<SnapshotItem>', () => {
    describe('when it renders', () => {
        let component;
        let icon = <svg><text>Icon</text></svg>;

        beforeAll(() => {
            const { output } = renderShallow(
                <SnapshotItem
                    tooltip="Revenue"
                    title="Revenue"
                    lineStyle="solid"
                    lineColor="red"
                    valueColor="green"
                    valueHint={ <p>17%</p> }
                    icon={ icon }
                >
                    $1.000.000
                </SnapshotItem>
            );

            component = output;
        });

        test('has proper markup', () => {
            expect(component).toMatchSnapshot();
        });
    });
});
