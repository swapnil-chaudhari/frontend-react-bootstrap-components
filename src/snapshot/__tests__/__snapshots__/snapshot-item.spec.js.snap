// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SnapshotItem> when it renders has proper markup 1`] = `
<div
  className="SnapshotItem"
>
  <div
    className="SnapshotItem-line"
    style={
      Object {
        "border": "1px solid red",
      }
    }
  />
  <div
    className="SnapshotItem-upper-icon"
  >
    <svg>
      <text>
        Icon
      </text>
    </svg>
  </div>
  <span
    className="SnapshotItem-title"
  >
    Revenue
  </span>
  <Help
    id="737884497096766615"
    message="Revenue"
    overlayType="tooltip"
    rootClose={false}
    tooltipPlacement="top"
    trigger={
      Array [
        "hover",
        "focus",
      ]
    }
  />
  <h2
    className="SnapshotItem-value"
    style={
      Object {
        "color": "green",
      }
    }
  >
    $1.000.000
  </h2>
  <p>
    17%
  </p>
</div>
`;
